marathon_acme/tests/test_sse_protocol.py

```diff
@@ -0,0 +1,285 @@
+from twisted.internet.error import ConnectionLost
+
+from marathon_acme.sse_protocol import SseProtocol
+
+
+class TestSseProtocol(TestCase):
+
+    def setUp(self):
+        super(TestSseProtocol, self).setUp()
+        self.protocol = SseProtocol()
+
+        class DummyTransport(object):
+            disconnecting = False
+        self.transport = DummyTransport()
+        self.protocol.transport = self.transport
+
+    def test_default_callback(self):
+        """
+        When data is received, followed by a blank line, the default callback,
+        'message', should be called with the data.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:hello\r\n\r\n')
+
+        self.assertThat(data, Equals(['hello']))
+
+    def test_multiline_data(self):
+        """
+        When multiple lines of data are specified in a single event, those
+        lines should be received by the callback with a '\n' character
+        separating them.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:hello\r\ndata:world\r\n\r\n')
+
+        self.assertThat(data, Equals(['hello\nworld']))
+
+    def test_different_newlines(self):
+        """
+        When data is received with '\r\n', '\n', or '\r', lines should be split
+        on those characters.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:hello\ndata:world\r\r\n')
+
+        self.assertThat(data, Equals(['hello\nworld']))
+
+    def test_empty_data(self):
+        """
+        When the data field is specified in an event but no data is given, the
+        callback should receive an empty string.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:\r\n\r\n')
+
+        self.assertThat(data, Equals(['']))
+
+    def test_no_data(self):
+        """
+        When the data field is not specified and the event is completed, the
+        callback should not be called.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'\r\n')
+
+        self.assertThat(data, Equals([]))
+
+    def test_space_before_value(self):
+        """
+        When a field/value pair is received, and there is a space before the
+        value, the leading space should be stripped.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data: hello\r\n\r\n')
+
+        self.assertThat(data, Equals(['hello']))
+
+    def test_space_before_value_strip_only_first_space(self):
+        """
+        When a field/value pair is received, and there is a space before the
+        value, the leading space should be stripped and the other spaces left
+        intact.
```

Comment:
I think "and there is a space" should rather be "and there are multiple spaces" for this test.